components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -82,7 +82,9 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
     ): GeckoResult<PromptResponse>? {
         val geckoResult = GeckoResult<PromptResponse>()
         val onConfirmSave: (Login) -> Unit = { login ->
-            geckoResult.complete(prompt.confirm(Autocomplete.LoginSelectOption(login.toLoginEntry())))
+            prompt.runIfNotCompleted {
+                geckoResult.complete(prompt.confirm(Autocomplete.LoginSelectOption(login.toLoginEntry())))
+            }
```

Comment:
I'm not sure if this helper is helping much? 🤔 

Without it, this line would be:

```Kotlin
if (!prompt.isComplete) {
   // ...
}
```

It's as long and isn't it easier to understand instead of a new function? 🤔

Reply:
Agree, it helps a bit for testing purpose as we only have to test the logic in one place, maybe we should be testing that individually.

Reply:
I updated the pr :)